spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -155,6 +155,7 @@ public void executeTask(String taskName, Map<String, String> taskDeploymentPrope
 					+ taskName);
 		}
 		taskExecutionRepository.updateExternalExecutionId(taskExecution.getExecutionId(), id);
+		return taskExecution.getExecutionId();
```

Comment:
This primary need for this executionId  is so that the ComposedTaskRunner knows what TaskExecution to monitor.  SCDF took ownership of creating the TaskExecutionId located [here](https://github.com/cppwfs/spring-cloud-dataflow/blob/45c6cff964474dc77620c3140205450c31ce132b/spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java#L143).  Any client that wants to monitor that TaskExecution has to get that taskExecutionId from Data Flow after task launch.